geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/LiveKafkaConsumerFeatureSource.scala

```diff
@@ -114,19 +124,86 @@ class LiveKafkaConsumerFeatureSource(entry: ContentEntry,
     }, 0, 1, TimeUnit.SECONDS)
   }
 
-  override def run(): Unit = while (running.get) {
-    queue.take() match {
-      case update: CreateOrUpdate => featureCache.createOrUpdateFeature(update)
-      case del: Delete            => featureCache.removeFeature(del)
-      case clr: Clear             => featureCache.clear()
-      case m                      => throw new IllegalArgumentException(s"Unknown message: $m")
+  override def run(): Unit =
+    while (running.get) {
+      queue.take() match {
+        case update: CreateOrUpdate =>
+          fireEvent(KafkaFeatureEvent.changed(this, update.feature))
```

Comment:
For updates and clears, no.

For deletes, since we decided on setting the bounds, and KDS delete control messages do not contain the feature info, we have to retrieve it from the featureCache.

Initially, when I had things in the other order, I couldn't get deletes to work in the listener, and I was confused.  After a little debugging, I found that the ordering matters.  Due to that, I decided to make the order consistent.